### PR TITLE
node pools: validate pool exists on job registration

### PR DESCRIPTION
**Pull request by tgross**
Add a new job admission hook for node pools that enforces the pool exists on registration. Also provide the skeleton hooks we need for Enterprise enforcement functions we'll implement later.

### CODE REVIEW

**Comment by tgross** [comment]
Note: when I merge this I'll need to fix up the missing `_ent.go` file in ENT, so I'll wait until I have time to land this and then immediately do that so the nightly OSS->ENT merge isn't broken.

**Review by lgfa29 — APPROVED** [review]

**Pull request merged**
